backend/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/ServerApiHelper.java

```diff
@@ -195,6 +197,9 @@ public static RuntimeException handleError(HttpClient.Response toBeClosed) {
       if (failedResponse.code() >= HttpURLConnection.HTTP_INTERNAL_ERROR) {
         return new ServerErrorException(formatHttpFailedResponse(failedResponse, null));
       }
+      if (failedResponse.code() == HTTP_TOO_MANY_REQUESTS) {
+        return new TooManyRequestsException("AI CodeFix usage has been capped. Too many requests have been made.");
```

Comment:
This method is generic, not only used for AI CodeFix, so I would suggest to use something a bit more generic, maybe only the 2nd sentence